app/src/routes/api/v1/gfw.router.ts

```diff
@@ -0,0 +1,248 @@
+            cloneUrl,
+        );
+        ctx.body = queryResults;
+    }
+
+    static async download(ctx: Context): Promise<void> {
+        try {
+            const format: string = ctx.query.format
+                ? (ctx.query.format as string)
+                : 'csv';
+            logger.debug('download format', format);
+            let mimetype: string;
+            switch (format) {
+                case 'csv':
+                    mimetype = 'text/csv';
+                    break;
+                case 'json':
+                default:
+                    mimetype = 'application/json';
+                    break;
+            }
+            const queryResults: Record<string, any> = await GfwService.download(
+                ctx.request.body.dataset.connectorUrl,
+                ctx.query.sql as string,
+                ctx.query.geostore_origin as string,
+                ctx.query.geostore_id as string,
+                format,
+            );
+            ctx.body = queryResults;
+            ctx.set(
+                'Content-disposition',
+                `attachment; filename=${ctx.request.body.dataset.id}.${format}`,
+            );
+            ctx.set('Content-type', mimetype);
+        } catch (err) {
+            ctx.body = ErrorSerializer.serializeError(
+                err.statusCode || 500,
+                err.error && err.error.error ? err.error.error[0] : err.message,
+            );
+            ctx.status = 500;
+        }
+    }
+}
+
+const serializeObjToQuery: (obj: Record<string, any>) => string = (
+    obj: Record<string, any>,
+) =>
+    Object.keys(obj)
+        .reduce((a, k) => {
+            a.push(`${k}=${encodeURIComponent(obj[k])}`);
+            return a;
+        }, [])
+        .join('&');
+
+const toSQLMiddleware: (ctx: Context, next: Next) => Promise<void> = async (
+    ctx: Context,
+    next: Next,
+) => {
+    const options: RequestToMicroserviceOptions & request.OptionsWithUri = {
+        method: 'GET',
+        json: true,
+        resolveWithFullResponse: true,
+        simple: false,
+        uri: '',
+    };
+    if (!ctx.query.sql && !ctx.request.body.sql) {
+        ctx.throw(400, 'sql or fs required');
+        return;
+    }
+
+    if (ctx.query.sql || ctx.request.body.sql) {
+        const params: Record<string, any> = { ...ctx.query, ...ctx.request.body };
+        options.uri = `${API_VERSION}/convert/sql2SQL?sql=${encodeURIComponent(
+            params.sql,
+        )}&experimental=true&raster=${ctx.request.body.dataset.type === 'raster'}`;
+        logger.debug(`Checking sql correct: ${params.sql}`);
+
+        if (params.geostore) {
+            options.uri += `&geostore=${params.geostore}`;
+        }
+        if (params.geojson) {
+            options.body = {
+                geojson: params.geojson,
+            };
+            options.method = 'POST';
+        }
+    } else {
+        logger.debug('Obtaining sql from featureService');
```

Comment:
test coverage shows that this else clause is not covered by tests. Is this needed, and used/supported by the GFW api?

Reply:
Removed feature service logic as it's not supported